clang/test/Analysis/stack-addr-ps.cpp

```diff
@@ -1,4 +1,9 @@
-// RUN: %clang_analyze_cc1 -analyzer-checker=core,debug.ExprInspection -verify %s -Wno-undefined-bool-conversion
+// RUN: %clang_analyze_cc1 \
+// RUN:   -analyzer-checker=core,debug.ExprInspection,unix.Malloc \
+// RUN:   -verify %s \
+// RUN:   -Wno-undefined-bool-conversion
+// unix.Malloc is necessary to model __builtin_alloca,
+// which could trigger an "unexpected region" bug in StackAddrEscapeChecker.
```

Comment:
Either make the `unix.Malloc` as a strong dependency of the `StackAddrEscapeChecker` (to get it enabled if it's enabled), or demonstrate that the `StackAddrEscapeChecker` would behave correctly (and not crash) if `unix.Malloc` is not enabled.

Reply:
Malloc / etc. modeling should not be a dependency of `StackAddrEscapeChecker` (and  in particular the `unix.Malloc` part of `MallocChecker` cannot be a strong dependency of anything, because it can generate reports, while only modeling checkers may be dependencies of other checkers). 

I don't think that we __need__ a separate testcase to see that `StackAddrEscapeChecker` does not crash on `__builtin_alloca` when `unix.Malloc` is not enabled -- the reason for this crash was that `unix.Malloc` introduced an unusual region type (`AllocaRegion`) and without that modeling the return value of `__builtin_alloca` is just a generic conjured region which should be already covered by the many other testcases. However, I'm not opposed to adding that testcase if you would like to see it.

Reply:
I added a run without malloc checker to demonstrate that there is no dependency see b2ed9f9